client/src/docs-ui/page/page.tsx

```diff
@@ -183,117 +183,119 @@ export class DocsPage {
     if (Build.isBrowser) {
       return (
         <Host class={pageStyle}>
-          <pageContext.Provider
-            state={{
-              selectedFilters: this.selectedFilters,
-              setSelectedFilters: this.setSelectedFilters,
-            }}
-          >
-            <docs-universal-nav
-              blend={this.blendUniversalNav}
-              heading="Amplify Docs"
-              brand-icon="/assets/logo-light.svg"
-              brand-icon-blend="/assets/logo-dark.svg"
-            />
-            {this.pageData && this.pageData.noTemplate
-              ? createVNodesFromHyperscriptNodes(this.pageData.body)
-              : [
-                  <docs-secondary-nav />,
-                  this.pageData && this.validFilterValue ? (
-                    <div class={sidebarLayoutStyle}>
-                      <amplify-toc-provider>
-                        <amplify-sidebar-layout>
-                          {this.showMenu() && (
-                            <amplify-sidebar-layout-sidebar
-                              slot="sidebar"
-                              top={this.sidebarStickyTop}
-                            >
-                              <docs-menu
-                                filterKey={this.filterKey}
-                                page={this.pageData}
-                                key={this.pageData?.productRootLink?.route}
-                              />
-                            </amplify-sidebar-layout-sidebar>
-                          )}
-                          {this.requiresFilterSelection() ? (
-                            <amplify-sidebar-layout-main slot="main">
-                              <amplify-toc-contents>
-                                {this.filterKey === "integration" ? (
-                                  <docs-choose-integration-anchor
-                                    page={this.pageData}
-                                  />
-                                ) : (
-                                  <docs-choose-anchor page={this.pageData} />
-                                )}
-                              </amplify-toc-contents>
-                            </amplify-sidebar-layout-main>
-                          ) : (
-                            [
-                              <amplify-sidebar-layout-main
-                                slot="main"
-                                class={mainStyle}
+          <amplify-block-switcher-provider>
```

Comment:
I dislike this pattern a lot. We should definitely not leak this information into the page component. Is there a reason we can't just fire a block-switcher tab switch event in which other block switchers are listening to?

Reply:
Is this still needed in the update to pageContext?

Reply:
Good catch! No it isn't